src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -282,6 +284,24 @@ int sd_dhcp_client_set_ifindex(sd_dhcp_client *client, int ifindex) {
         return 0;
 }
 
+int sd_dhcp_client_set_ifname(sd_dhcp_client *client, const char *ifname) {
+        assert_return(client, -EINVAL);
+        assert_return(ifname, -EINVAL);
+
+        if (!ifname_valid(ifname))
+                return -EINVAL;
+
+        strcpy(client->ifname, ifname);
```

Comment:
ifname_valid rejects names of length >= IFNAMSIZ. So the buffer seems to be too large by one.